### PR TITLE
py-espressopp: update to py3

### LINKED ISSUE #347: Update spack package
https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/py-espressopp/package.py

Still uses python2

### PR DESCRIPTION
**Pull request by junghans**
Fix https://github.com/espressopp/espressopp/issues/347

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-espressopp/package.py [hunk 1]
```diff
@@ -14,28 +14,26 @@ class PyEspressopp(CMakePackage):
        atomistic or bead-spring models as they are used in soft matter research
     """
     homepage = "https://espressopp.github.io"
-    url      = "https://github.com/espressopp/espressopp/tarball/v2.0.2"
+    url      = "https://github.com/espressopp/espressopp/tarball/v3.0.0"
     git      = "https://github.com/espressopp/espressopp.git"
 
-    version('develop', branch='master')
-    version('2.0.2',   sha256='8cf4525bca06426379f5b9fbb8cc2603f559d28a2e74d1d7694df963b8f3dc6c')
-    version('1.9.5',   sha256='8093f1a226f9fee8fb37c401767439a29ff3656dede3a44b4160169fc90d4d91')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-espressopp/package.py:22]: Old versions shouldn't be removed without first deprecating them: https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions

File: var/spack/repos/builtin/packages/py-espressopp/package.py [hunk 2]
```diff
@@ -14,28 +14,26 @@ class PyEspressopp(CMakePackage):
...
 
     depends_on("cmake@2.8:", type='build')
     depends_on("mpi")
-    depends_on("boost+serialization+filesystem+system+python+mpi cxxstd=11", when='@1.9.4:')
+    depends_on("boost+serialization+filesystem+system+python+mpi cxxstd=11")
     extends("python")
-    depends_on("python@2:2.8")
-    depends_on("py-mpi4py@2.0.0:", when='@1.9.4', type=('build', 'run'))
-    depends_on("py-mpi4py@1.3.1:", when='@1.9.4.1:', type=('build', 'run'))
+    depends_on("python@3:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-espressopp/package.py]: Build? Run? Link?

**Review by adamjstewart — COMMENTED** [review]
Where do you get the dependencies from? All I see is the `Pipfile(.lock)`.

File: var/spack/repos/builtin/packages/py-espressopp/package.py (hunk 1 above)
Comment (junghans) [reply var/spack/repos/builtin/packages/py-espressopp/package.py:22]: I really don't want to maintain a spackage for the ancient versions.

**Comment by junghans** [comment]
> Where do you get the dependencies from? All I see is the `Pipfile(.lock)`.

I looked at the CMake and then build it locally, I am not sure the `Pipfile` is really up to date.

File: var/spack/repos/builtin/packages/py-espressopp/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-espressopp/package.py:22]: That's fine, just deprecate them for now. If no one volunteers to maintain the old versions, you can remove them in the next Spack release.

Comment (junghans) [reply var/spack/repos/builtin/packages/py-espressopp/package.py:22]: I put them back, no too hard.

**Review by alalazo — APPROVED** [review]

**Pull request merged**